### PR TITLE
mgr/cephadm: get daemons by the status 

### PR DESCRIPTION
**Pull request by nizamial09**
after https://github.com/ceph/ceph/pull/59482

fetch the daemons by providing status or list of status.
based on the comment: https://github.com/ceph/ceph/pull/59482#discussion_r1743730014

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by epuertat — APPROVED** [review]
LGTM! Thanks for the improvement!

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -2419,7 +2419,8 @@ def list_daemons(self,
                      daemon_type: Optional[str] = None,
                      daemon_id: Optional[str] = None,
                      host: Optional[str] = None,
-                     refresh: bool = False) -> List[orchestrator.DaemonDescription]:
+                     refresh: bool = False,
+                     status: Optional[List[str]] = None) -> List[orchestrator.DaemonDescription]:
```
Comment (adk3798) [inline src/pybind/mgr/cephadm/module.py:2423]: These are supposed to matching `running`, `error`, `stopped` right? Do we want to validate whatever is passed in this list is actually a potential status? We know if someone was to pass `foo` or whatever for the status we're guaranteed to get no daemons.

Comment (nizamial09) [reply src/pybind/mgr/cephadm/module.py:2423]: yup, you are right. Is there an interface or enum which maintains all the possible daemon status?

Comment (adk3798) [reply src/pybind/mgr/cephadm/module.py:2423]: Yes, there is a `DaemonDescriptionStatus` class that lives in mgr/orchestrator/_interface.py that specifies them. You should be able to build a list of the status names to check against from that class. For example, in a little test script containing
```
import enum

from typing import Optional

class DaemonDescriptionStatus(enum.IntEnum):
    unknown = -2
    error = -1
    stopped = 0
    running = 1
    starting = 2  #: Daemon is deployed, but not yet running

    @staticmethod
    def to_str(status: Optional['DaemonDescriptionStatus']) -> str:
        if status is None:
            status = DaemonDescriptionStatus.unknown
        return {
            DaemonDescriptionStatus.unknown: 'unknown',
            DaemonDescriptionStatus.error: 'error',
            DaemonDescriptionStatus.stopped: 'stopped',
            DaemonDescriptionStatus.running: 'running',
            DaemonDescriptionStatus.starting: 'starting',
        }.get(status, '<unknown>')


print('foo' in [s.name for s in DaemonDescriptionStatus])
print('running' in [s.name for s in DaemonDescriptionStatus])
```
gives me
```
bash-5.2$ python3 daemon-status-get-valid-status.py 
False
True
```

**Pull request closed without merging**